testssl.sh

```diff
@@ -9186,41 +9608,60 @@ tls_sockets() {
-               check_tls_serverhellodone "$tls_hello_ascii"
-               hello_done=$?
-               [[ "$hello_done" -eq 3 ]] && process_full="ephemeralkey"
+               hello_done=1; skip=true
           fi
           for (( 1 ; hello_done==1; 1 )); do
-               sock_reply_file2=$(mktemp $TEMPDIR/ddreply.XXXXXX) || return 7
-               mv "$SOCK_REPLY_FILE" "$sock_reply_file2"
-
-               debugme echo -n "requesting more server hello data... "
-               socksend "" $USLEEP_SND
-               sockread_serverhello 32768
-
-               next_packet=$(hexdump -v -e '16/1 "%02X"' "$SOCK_REPLY_FILE")
-               next_packet="${next_packet%%[!0-9A-F]*}"
-               if [[ ${#next_packet} -eq 0 ]]; then
-                    # This shouldn't be necessary. However, it protects against
-                    # getting into an infinite loop if the server has nothing
-                    # left to send and check_tls_serverhellodone doesn't
-                    # correctly catch it.
-                    mv "$sock_reply_file2" "$SOCK_REPLY_FILE"
-                    hello_done=0
-               else
-                    tls_hello_ascii+="$next_packet"
-
-                    sock_reply_file3=$(mktemp $TEMPDIR/ddreply.XXXXXX) || return 7
-                    mv "$SOCK_REPLY_FILE" "$sock_reply_file3"
-                    mv "$sock_reply_file2" "$SOCK_REPLY_FILE"
-                    cat "$sock_reply_file3" >> "$SOCK_REPLY_FILE"
-                    rm "$sock_reply_file3"
+               if ! "$skip"; then
+                    sock_reply_file2=$(mktemp $TEMPDIR/ddreply.XXXXXX) || return 7
+                    mv "$SOCK_REPLY_FILE" "$sock_reply_file2"
+
+                    debugme echo -n "requesting more server hello data... "
+                    socksend "" $USLEEP_SND
+                    sockread_serverhello 32768
+
+                    next_packet=$(hexdump -v -e '16/1 "%02X"' "$SOCK_REPLY_FILE")
+                    next_packet="${next_packet%%[!0-9A-F]*}"
+
+                    if [[ ${#next_packet} -eq 0 ]]; then
+                         # This shouldn't be necessary. However, it protects against
+                         # getting into an infinite loop if the server has nothing
+                         # left to send and check_tls_serverhellodone doesn't
+                         # correctly catch it.
+                         mv "$sock_reply_file2" "$SOCK_REPLY_FILE"
+                         hello_done=0
+                    else
+                         tls_hello_ascii+="$next_packet"
 
-                    check_tls_serverhellodone "$tls_hello_ascii"
+                         sock_reply_file3=$(mktemp $TEMPDIR/ddreply.XXXXXX) || return 7
+                         mv "$SOCK_REPLY_FILE" "$sock_reply_file3"
+                         mv "$sock_reply_file2" "$SOCK_REPLY_FILE"
+                         cat "$sock_reply_file3" >> "$SOCK_REPLY_FILE"
+                         rm "$sock_reply_file3"
```

Comment:
don't know often this happens (length next packet >0) but file operations (incl. 1x mktemp) hurt performance.

Reply:
I looked over the code again, and I think we can eliminate all of these file operations, as it seems that the work this code is doing is only useful in debug mode. So, I think the solution is to change the code to only perform these operations if `$DEBUG` is at least 1. I'll make the change and submit a PR.